pkg/controller/privatelink/privatelink_controller.go

```diff
@@ -0,0 +1,241 @@
+		return errors.Wrap(err, "error watching cluster deprovision")
+	}
+
+	return nil
+}
+
+func (r *PrivateLinkReconciler) Reconcile(ctx context.Context, request reconcile.Request) (result reconcile.Result, returnErr error) {
+	logger := controllerutils.BuildControllerLogger(ControllerName, "clusterDeployment", request.NamespacedName)
+	logger.Debug("reconciling cluster deployment")
+	recobsrv := hivemetrics.NewReconcileObserver(ControllerName, logger)
+	defer recobsrv.ObserveControllerReconcileTime()
+
+	// Fetch the ClusterDeployment instance
+	cd := &hivev1.ClusterDeployment{}
+	err := r.Get(context.TODO(), request.NamespacedName, cd)
+	if apierrors.IsNotFound(err) {
+		logger.Debug("cluster deployment not found")
+		return reconcile.Result{}, nil
+	}
+	if err != nil {
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, errors.Wrap(err, "error getting ClusterDeployment")
+	}
+	logger = controllerutils.AddLogFields(controllerutils.MetaObjectLogTagger{Object: cd}, logger)
+
+	privateLink := &PrivateLink{
+		Client: r.Client,
+		cd:     cd,
+		logger: logger,
+	}
+
+	var privateLinkEnabled bool
+	var spokePlatformType configv1.PlatformType
+	switch {
+	case cd.Spec.Platform.GCP != nil:
+		spokePlatformType = configv1.GCPPlatformType
+		if cd.Spec.Platform.GCP.PrivateServiceConnect != nil {
+			privateLinkEnabled = cd.Spec.Platform.GCP.PrivateServiceConnect.Enabled
+		}
+	default:
+		logger.Debug("controller cannot service the clusterdeployment, so skipping")
+		return reconcile.Result{}, nil
+	}
+
+	privateLink.linkActuator, err = CreateActuator(r.Client, spokePlatformType, actuator.ActuatorTypeLink, cd, r.controllerconfig, logger)
+	if err != nil {
+		return reconcile.Result{}, errors.Wrap(err, "could not get link actuator")
+	}
+
+	// TODO: Determine and use hub platform type
+	// Ideally we would get the hub platform type from the cluster infrastructure. However,
+	// the hive service account does not have access. For now, we hard-code aws platform.
+	var hubPlatformType = configv1.AWSPlatformType
+
+	privateLink.hubActuator, err = CreateActuator(r.Client, hubPlatformType, actuator.ActuatorTypeHub, nil, r.controllerconfig, logger)
+	if err != nil {
+		return reconcile.Result{}, errors.Wrap(err, "could not get hub actuator")
+	}
+
+	return privateLink.Reconcile(privateLinkEnabled)
+}
+
+// CreateActuator creates an actuator based on the cloud platform and actuator type.
+func CreateActuator(
+	client client.Client,
+	platform configv1.PlatformType,
+	actuatorType actuator.ActuatorType,
+	cd *hivev1.ClusterDeployment,
+	config *hivev1.PrivateLinkConfig,
+	logger log.FieldLogger) (actuator.Actuator, error) {
+
+	switch platform {
+	case configv1.AWSPlatformType:
+		var awsConfig *hivev1.AWSPrivateLinkConfig
+		if config != nil {
+			awsConfig = config.AWS
+		}
+		if actuatorType == actuator.ActuatorTypeHub {
```

Comment:
It is not obvious how this falls back to the old AWSPrivateLink config -- please add a comment to that effect.

Reply:
[Done](https://github.com/openshift/hive/compare/b8f6a27707df3dd2dcd91175858fef744a0bc2f7..347334eb55af63da144fd48ed037f20948c499cf)